config/config_helpers.go

```diff
@@ -16,12 +17,13 @@ import (
 	"github.com/gruntwork-io/terragrunt/util"
 )
 
-var INTERPOLATION_PARAMETERS = `(\s*"[^"]*?"\s*,?\s*)*`
-var INTERPOLATION_SYNTAX_REGEX = regexp.MustCompile(fmt.Sprintf(`\$\{\s*\w+\(%s\)\s*\}`, INTERPOLATION_PARAMETERS))
+var INTERPOLATION_QUOTED_STRING = `"(?:\\.|[^\\"])*?"`
```

Comment:
The crux of the change is here.  Here's a snippet from regex101 to help explain the regex:
![image](https://user-images.githubusercontent.com/766820/59020959-93adb580-8810-11e9-9737-9d681ed1b574.png)

Reply:
In other words, `\\.` matches any escaped character, whereas `[^\\"]` matches anything except the backslash or the quote.  These are OR'd and then grouped in a non-capture group.  Finally, we use the kleene star to say we want zero or more of them (non-greedy).  Finally, the quotes are literal.